internal/template_test.go

```diff
@@ -43,254 +45,299 @@ var directories = []struct {
 	},
 }
 
-var files = []struct {
+type file struct {
 	workingDirectory string
 	name             string
 	template         string
-}{
-	{
-		workingDirectory: "test/rules/Test Rule ID",
-		name:             "main.rego",
-		template:         "templates/main.tpl.rego",
-	},
-	{
-		workingDirectory: "test/rules/Test Rule ID",
-		name:             "main_test.rego",
-		template:         "templates/main_test.tpl.rego",
-	},
-	{
-		workingDirectory: "test/rules/Test Rule ID/fixtures",
-		name:             "allowed.json",
-		template:         "templates/fixtures/allowed.json",
-	},
-	{
-		workingDirectory: "test/rules/Test Rule ID/fixtures",
-		name:             "denied1.yaml",
-		template:         "templates/fixtures/denied1.yaml",
-	},
-	{
-		workingDirectory: "test/rules/Test Rule ID/fixtures",
-		name:             "denied2.tf",
-		template:         "templates/fixtures/denied2.tf",
-	},
-	{
-		workingDirectory: "test/rules/Test Rule ID/fixtures",
-		name:             "denied.json.tfplan",
-		template:         "templates/fixtures/denied.json.tfplan",
-	},
-	{
-		workingDirectory: "test/lib",
-		name:             "main.rego",
-		template:         "templates/lib/main.tpl.rego",
-	},
-	{
-		workingDirectory: "test/lib/testing",
-		name:             "main.rego",
-		template:         "templates/lib/testing/main.tpl.rego",
-	},
-	{
-		workingDirectory: "test/lib/testing",
-		name:             "tfplan.rego",
-		template:         "templates/lib/testing/tfplan.tpl.rego",
-	},
+}
+
+func createTestFiles(format string) []file {
```

Comment:
There are two main changes to the tests:
1. Because the generated files differ based on the format flag, we need to have a different set of expected files per format. So the `createTestFiles` function takes care of that
2. Because of the different files per format, we have a test table with all supposed formats. The pre-existing tests in this file have been configured to run for each of those formats, so we check backwards compatibility for all formats